rsnapshot-program.pl

```diff
@@ -618,6 +618,30 @@ sub parse_config_file {
 			}
 		}
 
+		# Include from a directory
+		if ($var eq 'include_conf_dir') {
+			my $conf_pat = '';
+			if (defined($value) && -d $value && -r $value) {
+				# Check for a regex filter, default to .conf if not specified
+				if (defined($value2)) {
+					$conf_pat = $value2;
+				}
+				else {
+					$conf_pat = '.*\.conf$';
+				}
+				opendir(CONF_DIR, $value) or die $!;
+				while (my $conf_file = readdir(CONF_DIR)) {
```

Comment:
I believe that this will go all weird if any of the files in the directory themselves include an `include_conf_dir`, as when it recurses it will re-use the `CONF_DIR` handle. Please either test this to prove it doesn't happen, or read the entire directory into an array *and then close it* before processing the list of files, or use a lexically scoped handle.